fake_quant: move observer and fake_quant flags into buffers

vkuzo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#38368 fake_quant: move observer and fake_quant flags into buffers**

Summary:

There is a need for some customers to enable/disable these flags
in the middle of QAT.  To make it work properly with DDP,
we need to implement them using buffers so that they are replicated
properly to all the nodes.

This should solve issue https://github.com/pytorch/pytorch/issues/38081

Test Plan:

CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21537607](https://our.internmc.facebook.com/intern/diff/D21537607)

vkuzo committed:
fake_quant: move observer and fake_quant flags into buffers

Summary:

There is a need for some customers to enable/disable these flags
in the middle of QAT.  To make it work properly with DDP,
we need to implement them using buffers so that they are replicated
properly to all the nodes.

This should solve issue https://github.com/pytorch/pytorch/issues/38081

Test Plan:

CI

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 reviewed (approved):
this is probably not scriptable. but we haven't make fake quant module itself scriptable yet. so changes to make this scriptable will come later

vkuzo committed:
Update on "fake_quant: move observer and fake_quant flags into buffers"


Summary:

There is a need for some customers to enable/disable these flags
in the middle of QAT.  To make it work properly with DDP,
we need to implement them using buffers so that they are replicated
properly to all the nodes.

This should solve issue https://github.com/pytorch/pytorch/issues/38081

Test Plan:

CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21537607](https://our.internmc.facebook.com/intern/diff/D21537607)

[ghstack-poisoned]

vkuzo committed:
Update on "fake_quant: move observer and fake_quant flags into buffers"


Summary:

There is a need for some customers to enable/disable these flags
in the middle of QAT.  To make it work properly with DDP,
we need to implement them using buffers so that they are replicated
properly to all the nodes.

This should solve issue https://github.com/pytorch/pytorch/issues/38081

Test Plan:

CI

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21537607](https://our.internmc.facebook.com/intern/diff/D21537607)

[ghstack-poisoned]
